### PR TITLE
refactor(app): add modules step to protocol run setup

### LINKED ISSUE #8749: Setup tab - Step 2 (Modules)
# Overview
The second step in the protocol run details page's setup tab is related to module setup. 

# Design
- Step 2 accordion collapsed: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A101653  
- Step 2 accordion expanded: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A101653

# Implementation Details
- All that's needed from the user on this step is to place the required module(s) in the correct deck map location(s), connect them via USB, and power them on.
- If not modules are used in the protocol, this whole section should not be displayed

# Acceptance Criteria

- [ ] If no modules are used by this protocol, this whole section should not be displayed

- [ ] Required modules should be displayed in the correct deck map locations

- [ ] The Connection Status, Module type and generation, and the connected USB port should be displayed on the module representation on the deck map

- [ ] Clicking on "Proceed to Labware Setup" should collapse the Step 2 accordion and expand the Step 3 accordion

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

this adds the "Module Setup" step to the setup tab within the protocol run details page, and puts the `fetchModules` request within `useAttachedModules` on a 5 second polling interval.

closes #8749

# Changelog

 - Adds Modules step to Protocol Run Setup

# Review requests

check module setup deck map and module status for protocol runs with modules

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupModules.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  const heaterShakerModules = Object.values(
+    moduleRenderInfoForProtocolById
+  ).filter(module => module.moduleDef.model === 'heaterShakerModuleV1')
+
+  const moduleModels = map(
+    moduleRenderInfoForProtocolById,
+    ({ moduleDef }) => moduleDef.model
+  )
+
+  const hasADuplicateModule = new Set(moduleModels).size !== moduleModels.length
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/SetupModules.tsx:79]: maybe `hasADuplicatedModule`?

**Review by koji — APPROVED** [review]
Looks good to me

File: app/src/organisms/Devices/ProtocolRun/SetupModules.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/Devices/ProtocolRun/SetupModules.tsx:79]: i think either name is fine, @b-cooper any thoughts?

Comment (b-cooper) [reply app/src/organisms/Devices/ProtocolRun/SetupModules.tsx:79]: `hasADuplicateModule` follows existing variable naming patterns just fine, and effectively explains the purpose of the boolean.

**Review by b-cooper — APPROVED** [review]
Looks great 🏎️

**Pull request merged**